lib/utils.py

```diff
@@ -1,18 +1,21 @@
 from typing import List, Union
 
 
-def generate_ratios(num_steps: int, include_1: bool = True) -> List[float]:
+# maps line of 0-1 to a triangle from 0 to 1 back to 0
+def triangle(ratio: float) -> float:
```

Comment:
Hmm, this was pretty hard for me to figure out. 

Maybe this would be simpler:

```py
if ratio <= 0.5:
  return ratio * 2

assert ratio <= 1.0
return (1 - ratio) * 2
```

Reply:
Also I think a different func name would make this easier. 

```
def get_triangle_ratio(offset):
```

Reply:
yeah I think that is more readable, totally down.

The name is not my favorite.

`def line_to_triangle(offset):`

might be better. Down to change it though